### PR TITLE
Set TOPMOST flag on game data update to ensure we stay on top

### PR DESCRIPTION
**Pull request by OneXDeveloper**

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
Feels wrong that we have to spam it? Its not like we unset the flag at any point?

**Comment by misterokaygo** [comment]
> Feels wrong that we have to spam it? Its not like we unset the flag at any point? Is there a case where you found needing this?

I agree, if there is a use-case for this behavior it should be a setting `ForceAlwaysOnTop` or something.

**Comment by OneXDeveloper** [comment]
OK, I have changed it to only set it again when we become visible again. This should cutdown the spam and give the user an easy way to get it back on top (just toggle in game maps or change areas).

**Review by AudriusButkevicius — APPROVED** [review]

File: Overlay.cs [hunk 1]
```diff
@@ -146,12 +141,26 @@ private void MapUpdateTimer_Tick(object sender, EventArgs e)
...
+                }
                 mapOverlay.Refresh();
             }
 
             _timer.Start();
         }
+        
+        private void SetTopMost()
+        {
+            if (Map.AlwaysOnTop)
```
Comment (misterokaygo) [inline Overlay.cs:157]: I would move this check outside of the behavior and either nest the statements in the L144 block or putting another check un-nested on L149ish

```                if (!mapOverlay.Visible)
                {
                    mapOverlay.Show();
                    if (Map.AlwaysOnTop)
                    {
                         SetTopMost();
                     }
                }
                mapOverlay.Refresh();```

```                if (!mapOverlay.Visible)
                {
                    mapOverlay.Show();
                }
                    if (!mapOverlay.Visible && Map.AlwaysOnTop)
                    {
                         SetTopMost();
                     }
                mapOverlay.Refresh();```

or

Comment (OneXDeveloper) [reply Overlay.cs:157]: The way it’s written also stops it from spamming show when it already visible. We check the always on top setting in the set top most function.

Comment (misterokaygo) [reply Overlay.cs:157]: I think I didn't communicate well, I am asking you to move the setting check for `AlwaysOnTop` out of `SetTopMost` and move this into the Tick event, this separates the behavior of `SetTopMost` and business logic of the setting.

Comment (OneXDeveloper) [reply Overlay.cs]: Updated.

**Review by misterokaygo — APPROVED** [review]

File: Overlay.cs (hunk 1 above)
Comment (oviplokos) [reply Overlay.cs]: good afternoon, after an update from blizzard, the mapassist stopped working on d2r

**Pull request merged**